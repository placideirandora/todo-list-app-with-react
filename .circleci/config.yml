version: 2.1
jobs:
  build:
    docker:
      - image: circleci/node:10.16.0

    working_directory: ~/todo-list-app-with-react-js

    steps:
      - checkout
      - restore_cache:
          keys:
            - v1-dependencies-{{ checksum "package.json" }}
            - v1-dependencies-

      - run: npm install

      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-{{ checksum "package.json" }}

      - run: npm run test:coverage
