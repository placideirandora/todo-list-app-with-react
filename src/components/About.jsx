import React, { Fragment } from 'react';
import { aboutStyles } from '../styles/about';
import completeSvg from '../assets/images/complete';
import Image from './shared/Image';

const About = () => {
  return (
    <Fragment>
      <section style={aboutStyles.aboutWrapper}>
        <Image source={completeSvg} />
        <p>
          A web application that helps you create todos and keeps track of them
        </p>
      </section>
    </Fragment>
  );
};

export default About;
