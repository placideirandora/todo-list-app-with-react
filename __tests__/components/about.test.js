import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as adapter from '../config/enzyme.config';
import About from '../../src/components/About';

describe('About Component Test Cases', () => {
  it('should render the component without any failure', () => {
    const component = shallow(<About />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
