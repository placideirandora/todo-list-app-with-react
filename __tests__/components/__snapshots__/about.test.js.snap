// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component Test Cases should render the component without any failure 1`] = `
<Fragment>
  <section
    style={
      Object {
        "fontFamily": "Patua One",
        "fontSize": "2.6rem",
        "letterSpacing": "0.03rem",
        "paddingTop": "2rem",
        "textAlign": "center",
        "textTransform": "capitalize",
      }
    }
  >
    <Image
      height={300}
      source="complete.svg"
      width={400}
    />
    <p>
      A web application that helps you create todos and keeps track of them
    </p>
  </section>
</Fragment>
`;
